### PR TITLE
PLAT-10758: Remove the @pytest.mark.skip and tests fixes

### PR DESCRIPTION
**Pull request by symphony-youness**
### Ticket
[PLAT-10758](https://perzoinc.atlassian.net/browse/PLAT-10758)

### Description
Updated the generated models to take make the a pod_model>Product> sku property, optional
Removed the skip annotation from tests

Minor fixes in the PresenceService and related tests
Fixed forgotten object deserialization in the PresenceService tests
Removed the functions that deserializes JSON to python object in the resource utils

### Dependencies
List the other pull requests that should be merged before/along this one.

### Checklist
- [x] Referenced a ticket in the PR title and in the corresponding section
- [x] Filled properly the description and dependencies, if any
- [x] Unit tests updated or added
- [ ] Docstrings added or updated
- [ ] Updated the documentation in [docs folder](../docs)

### CODE REVIEW

File: templates/generate_client_api.sh [hunk 1]
```diff
@@ -37,4 +37,4 @@ generate_files() {
 generate_files https://raw.githubusercontent.com/symphonyoss/symphony-api-spec/master/agent/agent-api-public.yaml agent
 generate_files https://raw.githubusercontent.com/symphonyoss/symphony-api-spec/master/authenticator/authenticator-api-public.yaml auth
 generate_files https://raw.githubusercontent.com/symphonyoss/symphony-api-spec/master/login/login-api-public.yaml login
-generate_files https://raw.githubusercontent.com/symphonyoss/symphony-api-spec/master/pod/pod-api-public-deprecated.yaml pod
+generate_files https://raw.githubusercontent.com/symphonyoss/symphony-api-spec/master/pod/pod-api-public.yaml pod
```
Comment (symphony-elias) [inline templates/generate_client_api.sh:40]: @symphony-youness Why did you changed it ? If I remember well, some endpoints were missing in the non deprecated pod api

Comment (symphony-youness) [reply templates/generate_client_api.sh:40]: I thought it was a mistake 🤔

File: tests/core/service/presence/presence_service_test.py [hunk 2]
```diff
@@ -36,19 +41,19 @@ def fixture_presence_service(mocked_presence_api_client, auth_session):
...
+    mocked_presence_api_client.v2_user_presence_get.return_value = deserialize_object(V2Presence,
+                                                                                      "{\"category\":\"AVAILABLE\","
+                                                                                      "\"userId\":14568529068038,"
+                                                                                      "\"timestamp\":1533928483800}")
 
     presence = await presence_service.get_presence()
 
     assert presence.category == "AVAILABLE"
-    assert presence.userId == 14568529068038
+    assert presence.user_id == 14568529068038
```
Comment (symphony-youri) [inline tests/core/service/presence/presence_service_test.py:56]: is it related to the change here?

Comment (symphony-elias) [reply tests/core/service/presence/presence_service_test.py:56]: yes

File: templates/generate_client_api.sh (hunk 1 above)
Comment (symphony-elias) [reply templates/generate_client_api.sh:40]: I tried to recompile the java bdk with 20.12 specs and pod-api-public.yaml and it seems fine. So it is fine to use the pod-api-public.yaml

File: tests/core/service/presence/presence_service_test.py (hunk 2 above)
Comment (symphony-youness) [reply tests/core/service/presence/presence_service_test.py:56]: I didn't see catch the usage of this in the previous ticket.
The last commit removes all remaining usage of objects directly created from Json.

**Review by symphony-elias — APPROVED** [review]
LGTM

**Pull request merged**